### ISSUE TITLE
Undefined method '[]' for Options instance when using Capabilities instance and Headless option

### ISSUE THREAD

**Issue by drodriguezaltimetrik** [watir/watir#692]
## Meta -
Watir Version: 6.10
Selenium Version: 3.7.0

## Expected Behavior -
After merging :options => options_object and :headless => true to the Capabilities object, you should be able to open either Firefox/Chrome in headless mode

## Actual Behavior -
You get an error on Watir::Capabilities class, line 75: NoMethodError: undefined method '[]' for #<Selenium::WebDriver::Firefox::Options:0x007fdf91ac9ef0>

## Steps to reproduce -
1. Create a Capabilities instance for either Firefox or Chrome
1. Create an Options insteance for either Firefox or Chrome
1. Create the Watir instance with the merged capabilities: `Watir::Browser.new :chrome, capabilities.merge!({:options => options, :headless => true})`

**NOTE**:
The `.merge!` method on Capabilities will return a Hash object instead of a Capabilities object. This makes the Step 3 pass a Hash instead instead a Capability one to Watir.
The following code passes the Capabilities instance to Watir:

```
capabilities.merge!({:options => options, :headless => true})
Watir::Browser.new :chrome, capabilities
```

**Comment by jocubeit** [comment]
Any progress or comment on this... seems to be the same issue in 6.11.0?

**Comment by titusfortner** [comment]
Capabilities gave me some trauma in early 2017, and I've been completely avoiding them since. Finally getting around to addressing things.

This is definitely a bug; current code won't inject headless args if a Selenium Options instance is also being passed in.

This will be addressed in next Watir release. 

Only approach right now is to add the args to the Selenium Options class yourself:
```ruby
options.args << '--headless'
options.args << '--disable-gpu'
```

**Comment by titusfortner** [comment]
This has been fixed in 6.19, your original example will now work.
